vm/src/vm/runners/builtin_runner/ec_op.rs

```diff
@@ -62,31 +60,30 @@ impl EcOpBuiltinRunner {
         partial_sum: (Felt252, Felt252),
         doubled_point: (Felt252, Felt252),
         m: &Felt252,
-        alpha: &BigInt,
-        prime: &BigInt,
         height: u32,
-    ) -> Result<(BigInt, BigInt), RunnerError> {
-        let mut slope = felt_to_bigint(*m);
-        let mut partial_sum_b = (felt_to_bigint(partial_sum.0), felt_to_bigint(partial_sum.1));
-        let mut doubled_point_b = (
-            felt_to_bigint(doubled_point.0),
-            felt_to_bigint(doubled_point.1),
-        );
+    ) -> Result<(Felt252, Felt252), RunnerError> {
+        let mut slope = m.to_bigint();
+        let mut partial_sum_b = ProjectivePoint::from_affine(partial_sum.0, partial_sum.1)
+            .map_err(|_| RunnerError::PointNotOnCurve(Box::new(partial_sum)))?;
+        let mut doubled_point_b = ProjectivePoint::from_affine(doubled_point.0, doubled_point.1)
+            .map_err(|_| RunnerError::PointNotOnCurve(Box::new(doubled_point)))?;
         for _ in 0..height {
-            if (doubled_point_b.0.clone() - partial_sum_b.0.clone()).is_zero() {
-                #[allow(deprecated)]
+            if partial_sum_b.x() * doubled_point_b.z() == partial_sum_b.z() * doubled_point_b.x() {
                 return Err(RunnerError::EcOpSameXCoordinate(
                     Self::format_ec_op_error(partial_sum_b, slope, doubled_point_b)
                         .into_boxed_str(),
                 ));
             };
             if !(slope.clone() & &BigInt::one()).is_zero() {
```

Comment:
We could check the bit instead without creating any new temporary `BigInt`s (pseudo-code):
```rust
for i in 0..height.min(slope.bits()) {
    // check equality (does this need to happen several times?)
    if slope.bit(i) {
         partial_sum += doubled_point;
    }
    doubled_point = double_point.double(); // Is there an in-place version?
}
```